app/src/main/java/com/fastlink/zekrapp/appData/SingletonModule.kt

```diff
@@ -0,0 +1,24 @@
+package com.fastlink.zekrapp.appData
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object SingletonModule {
```

Comment:
No need for any of this, ZekirCategorySingleton and ZekirSingleton are already singletons, because in Kotlin any object is a singleton. So no need to provide a singleton instance here.

Reply:
Please first understand the theory behind dependency injection, then learn how it's applied in real applications.